src/overlays/actors/ovl_En_Ganon_Mant/z_en_ganon_mant.c

```diff
@@ -21,19 +27,444 @@ const ActorInit En_Ganon_Mant_InitVars = {
     (ActorFunc)EnGanonMant_Update,
     (ActorFunc)EnGanonMant_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ganon_Mant/EnGanonMant_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ganon_Mant/EnGanonMant_Destroy.s")
+static s16 sTearSizesMedium[] = {
+    0, 0, 0, 1, 1, 1, 1, 1, 0, 0,
+};
+
+static s16 sTearSizesLarge[] = {
+    0, 0, 0, 0, 1, 1, 2, 2, 2, 1, 1, 0, 0, 0, 0,
+};
+
+static s16 sTearSizesSmall[] = {
+    0, 0, 0, 0, 0, 0, 0,
+};
+
+typedef struct {
+    s16* tearAreaSizes;
+    s16 count;
+} TearShape; // size = 0x8
+
+/**
+ * The arrays pointed to by this table describe how many pixels should
+ * be removed from the cloak texture in a single pass
+ */
+static TearShape sTearShapes[] = {
+    { sTearSizesMedium, ARRAY_COUNT(sTearSizesMedium) },
+    { sTearSizesMedium, ARRAY_COUNT(sTearSizesMedium) },
+    { sTearSizesLarge, ARRAY_COUNT(sTearSizesLarge) },
+    { sTearSizesSmall, ARRAY_COUNT(sTearSizesSmall) },
+};
+
+/// How much each joint is affected by backwards/forwards swaying motion
+static f32 sBackSwayCoefficients[GANON_MANT_NUM_JOINTS] = {
+    0.0f, 1.0f, 0.5f, 0.25f, 0.1f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f,
+};
+
+static f32 D_80A24DB4[] = {
+    0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f,
+};
+
+/// How much each joint is affected by sideways swaying motion, tends to 0
+static f32 sSideSwayCoefficients[GANON_MANT_NUM_JOINTS] = {
+    0.0f, 1.0f, 0.9f, 0.8f, 0.7f, 0.6f, 0.5f, 0.4f, 0.3f, 0.2f, 0.1f, 0.0f,
+};
+
+static f32 D_80A24E00[] = {
+    0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f,
+};
+
+static f32 sDistMultipliers[GANON_MANT_NUM_JOINTS] = {
+    0.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.1f, 1.2f, 1.3f, 1.4f, 1.5f, 1.6f, 1.7f,
+};
+
+static f32 D_80A24E48[] = {
+    1.8f,
+};
```

Comment:
this should probably not be an array i guess?
the whole data situation here is very weird with all of these interruptions from unused/blank numbers

Reply:
I can make it not an array, I just felt it would be a little weird since all the others make more sense to be arrays, this one just so happens to only have one extra float.

Reply:
i guess i dont feel that the other stuff being arrays is enough to say that this was as well. same with the zeros being floats.
but in the end its not that important, so i dont have too strong of an opinion either way